File: day1.py
def frequency():
    freq = 0
    while True:
        op = input()
        print("Current frequency " + str(freq) + ", ")
        print("Change of " + op[0] + op[1:] + "; ")
        if op[0] == '+':
            freq += int(op[1:])
        elif op[0] == '-':
            freq -= int(op[1:])
        print("Resulting frequency: " + str(freq))
    return freq
    
frequency()

File: day1.cc
#include <iostream>
#include <string>
using namespace std;

int main() {
  int freq = 0;

  char op;
  int val;

  while (cin >> op) {
    cout << "Current frequency " << freq << ", ";
    cin >> val;
    cout << "change of " << op << val << "; ";
    if (op == '+') {
      freq += val;
    }
    else if (op == '-') {
      freq -= val;
    }
    cout << "resulting frequency " << freq << endl;
  }
  return 0;
}
